api/api.go

```diff
@@ -44,6 +45,7 @@ type API interface {
 // DownloadsGenerator pre generates full file downloads for the specified dataset/edition/version
 type DownloadsGenerator interface {
 	Generate(ctx context.Context, datasetID, instanceID, edition, version string) error
+	GetProducer() download.KafkaProducer
```

Comment:
We can delete these functions we added that we didn't use in the end. I'll mark the rest throughout